src/wmtk/EdgeCollapsingConn.cpp

```diff
@@ -4,18 +4,140 @@
 #include <wmtk/TetMesh.h>
 
 #include <wmtk/utils/TupleUtils.hpp>
+#include <wmtk/utils/VectorUtils.h>
+
+#include <spdlog/spdlog.h>
+
+#include <algorithm>
+#include <cstddef>
+#include <limits>
+#include <type_traits>
 
 bool wmtk::TetMesh::collapse_edge(const Tuple& loc0, std::vector<Tuple>& new_edges)
 {
     if (!collapse_before(loc0)) return false;
 
+    auto link_condition = [&VC = this->m_vertex_connectivity,
```

Comment:
they should be functions inside the class :)

Reply:
It captures a `this`

Reply:
I would prefer to refactor once it is needed.

Reply:
Maybe create an issue like "unify 2d and 3d link condition etc."